src/foam/core/AbstractFObject.java

```diff
@@ -144,4 +144,29 @@ public boolean hasDefaultValue(String prop) {
       return null;
     }
   }
+
+  public byte[] sign(PrivateKey key) {
+    return this.sign("SHA256withRSA", key);
+  }
+
+  public byte[] sign(String algorithm, PrivateKey key) {
+    try {
+      Signature signature = Signature.getInstance(algorithm);
```

Comment:
How expensive are these two lines? Can we reuse these by just passing in the Signature?

Reply:
It didn't make an effect in terms of performance.